stdlib/REPL/src/REPL.jl

```diff
@@ -1492,60 +1498,63 @@ function banner(io::IO = stdout; short = false)
-            commit_string = "$(branch)/$(commit) (fork: $(distance) commits, $(days) $(unit))"
+            styled"""{emphasis:$branch}/{grey:$commit} \
+                     ({italic:{success:{bold,(slant=normal):↑} $distance commits}, \
+                     {warning:{(slant=normal):⌛} $days $unit}})"""
         end
     end
 
-    commit_date = isempty(Base.GIT_VERSION_INFO.date_string) ? "" : " ($(split(Base.GIT_VERSION_INFO.date_string)[1]))"
-
-    if get(io, :color, false)::Bool
-        c = Base.text_colors
-        tx = c[:normal] # text
-        jl = c[:normal] # julia
-        d1 = c[:bold] * c[:blue]    # first dot
-        d2 = c[:bold] * c[:red]     # second dot
-        d3 = c[:bold] * c[:green]   # third dot
-        d4 = c[:bold] * c[:magenta] # fourth dot
-
-        if short
-            print(io,"""
-              $(d3)o$(tx)  | Version $(VERSION)$(commit_date)
-             $(d2)o$(tx) $(d4)o$(tx) | $(commit_string)
-            """)
-        else
-            print(io,"""               $(d3)_$(tx)
-               $(d1)_$(tx)       $(jl)_$(tx) $(d2)_$(d3)(_)$(d4)_$(tx)     |  Documentation: https://docs.julialang.org
-              $(d1)(_)$(jl)     | $(d2)(_)$(tx) $(d4)(_)$(tx)    |
-               $(jl)_ _   _| |_  __ _$(tx)   |  Type \"?\" for help, \"]?\" for Pkg help.
-              $(jl)| | | | | | |/ _` |$(tx)  |
-              $(jl)| | |_| | | | (_| |$(tx)  |  Version $(VERSION)$(commit_date)
-             $(jl)_/ |\\__'_|_|_|\\__'_|$(tx)  |  $(commit_string)
-            $(jl)|__/$(tx)                   |
-
-            """)
-        end
-    else
-        if short
-            print(io,"""
-              o  |  Version $(VERSION)$(commit_date)
-             o o |  $(commit_string)
-            """)
-        else
-            print(io,"""
-                           _
-               _       _ _(_)_     |  Documentation: https://docs.julialang.org
-              (_)     | (_) (_)    |
-               _ _   _| |_  __ _   |  Type \"?\" for help, \"]?\" for Pkg help.
-              | | | | | | |/ _` |  |
-              | | |_| | | | (_| |  |  Version $(VERSION)$(commit_date)
-             _/ |\\__'_|_|_|\\__'_|  |  $(commit_string)
-            |__/                   |
-
-            """)
-        end
+    commit_date = isempty(Base.GIT_VERSION_INFO.date_string) ? "" : styled" {light:($(split(Base.GIT_VERSION_INFO.date_string)[1]))}"
+    doclink = styled"{bold:Documentation:} {(underline=grey),link={https://docs.julialang.org}:https://docs.julialang.org}"
+    help = styled"Type {region,repl_prompt_help:?} for help, {region,repl_prompt_pkg:]?} for {(underline=grey),link={https://pkgdocs.julialang.org/}:Pkg} help."
+
+    size = min(if     all(displaysize(io) .>= (5, 70)); 4 # Full size
+               elseif all(displaysize(io) .>= (5, 45)); 3 # Narrower
+               elseif all(displaysize(io) .>= (3, 50)); 2 # Tiny
+               else 1 end,
+               max(0, maxsize))
```

Comment:
IIUC currently the banner size is always adaptive and the min of what fits on screen and what a user asked for. I think auto being adaptive is good. However, if I ask for `julia --banner=7`, I think I should get banner 7 regardless of screen size.

Reply:
Hmmm, currently `--banner` operates as setting the upper limit on the banner size. I see this as sensible behaviour, have you got any thoughts on how both could be possible?

Reply:
I also struggle to imagine a situation when it would be desirable to have a banner that doesn't fit on the current screen _without_ text wrapping (breaking the layout).